src/components/DashboardJoinCard/DashboardJoinCardPresentation.js

```diff
@@ -65,7 +65,9 @@ function DashboardJoinCardPresentation({
               <select
                 id="role-to-assign"
                 name="role-to-assign"
-                onChange={(e) => setSelectedRole(e.target.value)}
+                onChange={(e) =>
+                  setSelectedRole(e.target.value.split(' ').join('_'))
```

Comment:
Looks good for now.
In the long term we should be moving towards the discussed solution of having a mapping of displayable values and BE expected keys.